Support 16 KiB Pages on AArch64 Targets

hmelder opened the pull request:
This PR adds support for 16 KiB pages on AArch64. `libobjc2` was also broken on Asahi Linux because of this. Additionally, starting November 1st, 2025, Google requires all applications targeting Android 15+ to support 16 KiB page sizes. 

All hard-coded page sizes in `blocks_to_imp.c` were replaced by one call to `getpagesize()` in the initialization function.
PowerPC CI will probably fail, as the CI is still misconfigured and reports 4 KiB pages.

davidchisnall commented on shims.h:
```diff
@@ -0,0 +1,21 @@
+
+#else // Assume that the platform implements the X/Open System Interface
+
+#include <unistd.h>
```
`getpagesize` is not a POSIX function.  I think you want:

```
long getpagesize(void)
{
    return sysconf(_SC_PAGESIZE);
}
```

davidchisnall commented on shims.c:
```diff
@@ -0,0 +1,20 @@
+
+#include "safewindows.h"
+
+int getpagesize(void) {
```
This API is also not standard, so we could just put this with the platform-specific code at the top of the one file that needs it.

hmelder commented:
Sorry for the delay. I've moved the definition of `pagesize()` into `block_to_imp.c` and removed the dependency on the non-standard `getpagesize()` function. The changes were tested on Asahi Linux after the rebase.

hmelder commented:
Oops, I still need to fix the PPC CI.

hmelder commented:
Turns out, changing the page size of the qemu emulator is apparently not possible, so we either need to disable the CI for powerpc64le, or implement support for 4 KiB pages.

```c
#include <unistd.h>
#include <stdio.h>

int main(void) {
	printf("The page size of the system is %d bytes.\n", getpagesize());
	return 0;
}
```

```sh
qemu-ppc64le: warning: Deprecated page size option cannot change host page size (4096)
The page size of the system is 4096 bytes
```

hmelder commented:
@davidchisnall the Cirrus CI is not executing due to insufficient compute credits.

davidchisnall commented:
Cirrus is a bit flaky here, happy to merge if you clean up the history.
